Assets/Scripts/Skills/Skill_Mage.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu(fileName = "Mage Skill", menuName = "New Mage Skills", order = 1)]
+public class Skill_Mage : Skill {
+
+    InputHandlerBuilder builder;
+    InputHandler ih;
+
+    MonoBehaviour m;
+
+    public override bool CanCast(Stats stats)
+    {
+        // Check if current health >= max health as exemeple
+        bool validate = stats.GetHealth() >= stats.GetMaxHealth() && !isCooldown && !isCasting;
+
+        return validate;
+    }
+
+    public override bool CanCast(PlayerAttack pInfo)
+    {
+        return cost == pInfo.GetPassifVal() && !isCooldown && !isCasting;
+    }
+
+    public override IEnumerator Cast()
+    {
+        isCasting = true;
+
+        castStartTime = Time.time;
+        
+        // deux conditions : soit push la touche, soit push jusqua push le max time
+        yield return new WaitUntil(() => !ih.FirstSkill() && Time.time - castStartTime < castTime || Time.time - castStartTime >= minCastTime);
+
+        isCasting = false;
+
+        m.StartCoroutine(OnCooldown());
+
```

Comment:
https://answers.unity.com/questions/275669/how-can-i-use-coroutines-in-scriptableobject.html
Je pense que ca peux etre une bonne idée un truc du genre
Mais la glisser un gameobject en parametre du scriptable object, jpense que c'est pas ouf

Reply:
Nan mais mec, on s'en fou tout les scripts sont des monobehaviour, quand on lis le nom de la variable on veux savoir a quoi elle sert
On sait deja que c'est un monobehaviour via son type.
Si c'est le gameobject a spawn , tu l'appelle spawnPrefab ou un truc du genre